pypeit/core/datacube.py

```diff
@@ -1270,11 +1290,47 @@ def coadd_cube(files, opts, spectrograph=None, parset=None, overwrite=False):
                     msgs.warn("Scale correction will not be performed")
                 relScaleImg = relScaleImgDef.copy()
 
+        # Set the default behaviour if a global skysub frame has been specified
+        this_skysub = skysub_default
+        if skysub_default == "image":
+            skysubImg = spec2DObj.skymodel
+        else:
+            skysubImg = skysubImgDef.copy() * exptime
+        # See if there's any changes from the default behaviour
+        if opts['skysub_frame'][ff] is not None:
+            if opts['skysub_frame'][ff].lower() == 'default':
+                if skysub_default == "image":
+                    skysubImg = spec2DObj.skymodel
+                    this_skysub = "image"  # Use the current spec2d for sky subtraction
+                else:
+                    skysubImg = skysubImgDef.copy() * exptime
+                    this_skysub = skysub_default  # Use the global value for sky subtraction
+            elif opts['skysub_frame'][ff].lower() == 'image':
+                skysubImg = spec2DObj.skymodel
+                this_skysub = "image"  # Use the current spec2d for sky subtraction
+            elif opts['skysub_frame'][ff].lower() == 'none':
+                skysubImg = np.array([0.0])
+                this_skysub = "none"  # Don't do sky subtraction
+            else:
+                try:
+                    # Load a user specified frame for sky subtraction
+                    msgs.info("Loading skysub frame:"+msgs.newline()+opts['skysub_frame'][ff])
+                    spec2DObj_sky = spec2dobj.Spec2DObj.from_file(opts['skysub_frame'][ff], detname)
```

Comment:
Same comment here about limiting how many lines are in the try block.

Reply:
Good point - fixed!